app/controllers/report_controller/widgets.rb

```diff
@@ -655,7 +655,17 @@ def widget_set_record_vars(widget)
     #      end
     #      widget.miq_widget_shortcuts = ws
     #    end
-    widget.options[:url] = @edit[:new][:url] unless @edit[:new][:url].blank?
+    if @sb[:wtype] == "rf"
+      if @edit[:new][:feed_type] == "internal"
+        widget.resource = RssFeed.find(@edit[:new][:rss_feed_id]) if @edit[:new][:rss_feed_id]
```

Comment:
For instance when editing widget and changing from external to internal, we may get `nil` here.

Maybe there are more cases like this. My point was, I am editing code which lacks test coverage. Let's be extra careful.